[Bug] - Nil pointer exception

```go
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x48721c3]

goroutine 187 [running]:
github.com/goccy/go-json/internal/encoder/vm.Run(0xc002f276c0, {0xc001d27c00, 0xc001070180, 0x0}, 0x0)
        /Users/sujit/go/pkg/mod/github.com/goccy/go-json@v0.7.8/internal/encoder/vm/vm.go:197 +0x2f5a3
github.com/goccy/go-json.encodeRunCode(0x4de31c0, {0xc001d27c00, 0x6806d90, 0xc0009bf580}, 0xc00106e000)
        /Users/sujit/go/pkg/mod/github.com/goccy/go-json@v0.7.8/encode.go:307 +0x68
github.com/goccy/go-json.encode(0xc002f276c0, {0x4de31c0, 0xc0011cb7d0})
        /Users/sujit/go/pkg/mod/github.com/goccy/go-json@v0.7.8/encode.go:232 +0x21c
github.com/goccy/go-json.marshal({0x4de31c0, 0xc0011cb7d0}, {0x0, 0x0, 0x40169aa})
        /Users/sujit/go/pkg/mod/github.com/goccy/go-json@v0.7.8/encode.go:147 +0xba
github.com/goccy/go-json.MarshalWithOption(...)
        /Users/sujit/go/pkg/mod/github.com/goccy/go-json@v0.7.8/json.go:186
github.com/goccy/go-json.Marshal({0x4de31c0, 0xc0011cb7d0})
        /Users/sujit/go/pkg/mod/github.com/goccy/go-json@v0.7.8/json.go:171 +0x2a
```

We just ran into this. Here is my reproducer which looks similar to what @orisano reported. What i find interesting is that an empty interface declaration does not trigger the problem: You need an interface with a method.

```go
package main

import (
	"encoding/json"
	"fmt"

	goccy "github.com/goccy/go-json"
)

func main() {
	type Empty interface{}
	type NonEmpty interface {
		A()
	}

	type SomeStruct struct {
		Empty    Empty
		NonEmpty NonEmpty // Comment this line and you will receive no panic.
	}

	originalEncoded, originalError := json.Marshal(SomeStruct{}) // Works
	fmt.Printf("Original: %s\n-> %s\n", originalEncoded, originalError)

	goccyEncoded, goccyError := goccy.Marshal(SomeStruct{}) // Panics
	fmt.Printf("Original: %s\n-> %s\n", goccyEncoded, goccyError)
}
```

```
Original: {"Empty":null,"NonEmpty":null}
-> %!s(<nil>)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x5104e3]

goroutine 1 [running]:
github.com/goccy/go-json/internal/encoder/vm.Run(0xc000102dd0, {0xc000028400, 0x0, 0x400}, 0xc0000320c0)
        src/github.com/goccy/go-json/internal/encoder/vm/vm.go:197 +0x2f5a3
github.com/goccy/go-json.encodeRunCode(0x524c20, {0xc000028400, 0x65fda8, 0xc000032080}, 0xc000028800)
        src/github.com/goccy/go-json/encode.go:307 +0x68
github.com/goccy/go-json.encode(0xc000102dd0, {0x524c20, 0xc00006a060})
        src/github.com/goccy/go-json/encode.go:232 +0x21c
github.com/goccy/go-json.marshal({0x524c20, 0xc00006a060}, {0x0, 0x0, 0xc00005df50})
        src/github.com/goccy/go-json/encode.go:147 +0xba
github.com/goccy/go-json.MarshalWithOption(...)
        src/github.com/goccy/go-json/json.go:186
github.com/goccy/go-json.Marshal({0x524c20, 0xc00006a060})
        src/github.com/goccy/go-json/json.go:171 +0x2a
main.main()
        tt.go:24 +0x10a
exit status 2
```

@goccy hope that helps!

--- a/encode_test.go
+++ b/encode_test.go
@@ -2148,3 +2148,23 @@ func TestEmptyStructInterface(t *testing.T) {
 		t.Fatalf("failed to encode empty struct interface. expected:[%q] but got:[%q]", expected, got)
 	}
 }
+
+func TestIssue290(t *testing.T) {
+	type Issue290 interface {
+		A()
+	}
+	var a struct {
+		A Issue290
+	}
+	expected, err := stdjson.Marshal(a)
+	if err != nil {
+		t.Fatal(err)
+	}
+	got, err := json.Marshal(a)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !bytes.Equal(expected, got) {
+		t.Fatalf("failed to encode non empty interface. expected = %q but got %q", expected, got)
+	}
+}

--- a/internal/cmd/generator/vm.go.tmpl
+++ b/internal/cmd/generator/vm.go.tmpl
@@ -194,7 +194,9 @@ func Run(ctx *encoder.RuntimeContext, b []byte, codeSet *encoder.OpcodeSet) ([]b
 			if code.Flags&encoder.NonEmptyInterfaceFlags != 0 {
 				iface := (*nonEmptyInterface)(up)
 				ifacePtr = iface.ptr
-				typ = iface.itab.typ
+				if iface.itab != nil {
+				    typ = iface.itab.typ
+				}
 			} else {
 				iface := (*emptyInterface)(up)
 				ifacePtr = iface.ptr

--- a/internal/encoder/vm/vm.go
+++ b/internal/encoder/vm/vm.go
@@ -194,7 +194,9 @@ func Run(ctx *encoder.RuntimeContext, b []byte, codeSet *encoder.OpcodeSet) ([]b
 			if code.Flags&encoder.NonEmptyInterfaceFlags != 0 {
 				iface := (*nonEmptyInterface)(up)
 				ifacePtr = iface.ptr
-				typ = iface.itab.typ
+				if iface.itab != nil {
+					typ = iface.itab.typ
+				}
 			} else {
 				iface := (*emptyInterface)(up)
 				ifacePtr = iface.ptr

--- a/internal/encoder/vm_color/vm.go
+++ b/internal/encoder/vm_color/vm.go
@@ -194,7 +194,9 @@ func Run(ctx *encoder.RuntimeContext, b []byte, codeSet *encoder.OpcodeSet) ([]b
 			if code.Flags&encoder.NonEmptyInterfaceFlags != 0 {
 				iface := (*nonEmptyInterface)(up)
 				ifacePtr = iface.ptr
-				typ = iface.itab.typ
+				if iface.itab != nil {
+					typ = iface.itab.typ
+				}
 			} else {
 				iface := (*emptyInterface)(up)
 				ifacePtr = iface.ptr

--- a/internal/encoder/vm_color_indent/vm.go
+++ b/internal/encoder/vm_color_indent/vm.go
@@ -194,7 +194,9 @@ func Run(ctx *encoder.RuntimeContext, b []byte, codeSet *encoder.OpcodeSet) ([]b
 			if code.Flags&encoder.NonEmptyInterfaceFlags != 0 {
 				iface := (*nonEmptyInterface)(up)
 				ifacePtr = iface.ptr
-				typ = iface.itab.typ
+				if iface.itab != nil {
+					typ = iface.itab.typ
+				}
 			} else {
 				iface := (*emptyInterface)(up)
 				ifacePtr = iface.ptr

--- a/internal/encoder/vm_indent/vm.go
+++ b/internal/encoder/vm_indent/vm.go
@@ -194,7 +194,9 @@ func Run(ctx *encoder.RuntimeContext, b []byte, codeSet *encoder.OpcodeSet) ([]b
 			if code.Flags&encoder.NonEmptyInterfaceFlags != 0 {
 				iface := (*nonEmptyInterface)(up)
 				ifacePtr = iface.ptr
-				typ = iface.itab.typ
+				if iface.itab != nil {
+					typ = iface.itab.typ
+				}
 			} else {
 				iface := (*emptyInterface)(up)
 				ifacePtr = iface.ptr

